child_sync_typo3/wizard/child_depart_wizard.py

```diff
@@ -0,0 +1,28 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino <ecino@compassion.ch>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+from openerp.osv import orm
+from ..model.sync_typo3 import Sync_typo3
+
+
+class end_sponsorship_wizard(orm.TransientModel):
+    _inherit = 'end.sponsorship.wizard'
+
+    def child_depart(self, cr, uid, ids, context=None):
+        res = super(end_sponsorship_wizard, self).child_depart(
+            cr, uid, ids, context)
+
+        wizard = self.browse(cr, uid, ids[0], context)
+        child = wizard.child_id
+
+        if child.state == 'I':

```

Comment:
This is not working, because you are writing the state of the child to 'F' in the super method. You have to perform this part before calling the super method.